src/FISTA.jl

```diff
@@ -162,6 +165,14 @@ function iterate(solver::FISTA, iteration::Int=0)
   # proximal map
   solver.reg.prox!(solver.x, solver.regFac*solver.ρ*solver.reg.λ; solver.reg.params...)
 
+  # gradient restart conditions
+  if solver.restart == :gradient
+    if real(solver.res ⋅ (solver.x .- solver.xᵒˡᵈ) ) > 0 #if momentum is at an obtuse angle to the negative gradient
```

Comment:
fyi, one can make this "0" a parameter for more flexibility:
https://github.com/JeffFessler/MIRT.jl/blob/49ca5c809dcde60dac1bc33d3cbde21760664b11/src/algorithm/general/pogm_restart.jl#L44
in practice we generally use the default anyway...

Reply:
Got it, I'll incorporate that as an option